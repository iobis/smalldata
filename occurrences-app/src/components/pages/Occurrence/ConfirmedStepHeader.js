import PropTypes from 'prop-types'
import React from 'react'
import StepHeader from './StepHeader'

export default function ConfirmedStepHeader({ dataDescription, onStepTitleClick, selectedData, stepDescription, stepTitle }) {
  return (
    <StepHeader
      className="confirmed has-background-white has-text-black"
      iconVisible={true}
      onStepTitleClick={onStepTitleClick}
      dataDescription={dataDescription}
      selectedData={selectedData}
      stepDescription={stepDescription}
      stepTitle={stepTitle}/>
  )
}

ConfirmedStepHeader.propTypes = {
  dataDescription:  PropTypes.string.isRequired,
  onStepTitleClick: PropTypes.func.isRequired,
  selectedData:     PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
  stepDescription:  PropTypes.string.isRequired,
  stepTitle:        PropTypes.string.isRequired
}
