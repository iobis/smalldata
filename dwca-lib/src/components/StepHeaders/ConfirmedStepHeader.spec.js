import ConfirmedStepHeader from './ConfirmedStepHeader'
import React from 'react'
import renderer from 'react-test-renderer'

describe('ActiveStepHeader', () => {
  it('renders correctly', () => {
    expect(renderer.create(
      <ConfirmedStepHeader
        dataDescription="data description"
        onStepTitleClick={() => {}}
        selectedData="selected data"
        stepDescription="step description"
        stepTitle="step title"/>
    ).toJSON()).toMatchSnapshot()
  })
})
