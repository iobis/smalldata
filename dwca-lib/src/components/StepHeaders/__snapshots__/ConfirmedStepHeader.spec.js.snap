// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveStepHeader renders correctly 1`] = `
<header
  className="step-header columns is-vcentered confirmed has-background-white has-text-black"
  onClick={[Function]}
>
  <div
    className="column is-1"
  >
    <p
      className="datadescription"
    >
      data description
    </p>
  </div>
  <p
    className="selected-data column"
  >
    selected data
  </p>
  <div
    className="column details"
  >
    <h3
      className="is-size-6"
    >
      step description
    </h3>
    <h2
      className="step-title"
    >
      <i
        className="check-circle"
        icon="check-circle"
      />
      step title
    </h2>
  </div>
</header>
`;
